core/templates/dev/head/pages/base.html

```diff
@@ -213,6 +213,7 @@ <h2>
     <script src="/templates/dev/head/components/StoryCreationService.js"></script>
 
     <script src="/templates/dev/head/components/CkEditorRteDirective.js"></script>
+    <script src="/templates/dev/head/components/CkEditorWidgets.js"></script>
```

Comment:
This should be only imported in pages where CKEditor is actually needed. Look at pages where CKEditor is used and move the script-imports related to CKEditor there, also move the script-import for the whole CKEditor together with it so we load it only in places where we need it https://github.com/oppia/oppia/blob/626c4fd98fc44f788d7c7642eb51b8e1362c86dc/core/templates/dev/head/pages/header_js_libs.html#L16

Reply:
ok

Reply:
ok

Reply:
done

Reply:
Can you give hint for finding where editor is used, as ``` git grep 'ck-editor-rte' ``` gives only 3 files

Reply:
It might be the case that it is only used in these three places.

Reply:
Also the script-imports shouldn't be included in directive HTML files but only in page HTML files.

Reply:
Hii @vojtechjelinek , but ``` git grep ``` shows only directive html file, it shouldn't show any page HTML.
I go through the files and found  ``` <ck-editor-rte> ``` is used in other directives like ``` schema_based_html_editor_directive ```.
Thats the main reason i import scripts in base.html file

Reply:
Yes, but these directives are used only in some pages, so you need to find these pages and put the script-imports there.

Reply:
Ohh, thanks for the help

Reply:
Sir i go through the suggestion you give

> Yes, but these directives are used only in some pages, so you need to find these pages and put the script-imports there.

Reply:
Okay, but you should be able to move the `RteHelperService.js` from `base.html` to individual page file too and also if you look fro example at https://github.com/oppia/oppia/blob/4a57b0d214966a1e2447a0e4d8c9ddd014432b78/core/templates/dev/head/pages/about/about.html#L605 the `{{ super() }}` means that all the script-imports from `base.html` will be included in this page too. 

Basically, you need to move every file you have that is used to mainly implement CKEditor, the CKEditor of course depends on more general files that are used everywhere and these files can be located in `base.html` since then they will be loaded on every page.

Reply:
Ok, i will try

Reply:
@vojtechjelinek , i go through the code base and here i found where scripts for edior is used
![imageedit_1_7577504973](https://user-images.githubusercontent.com/34139754/53055420-a30d2b00-34cd-11e9-98f2-ea384e230d2d.png)

As I don't know all about pages, so can you tell me where there is no need of importing editor scripts.
From the above, I think pages are ``` admin.html ```, ``` creator_dashboard.html ```. I know only about these pages 
And where there is need is ``` exploration_editor.html ```.

Reply:
It is probably used in every page in the list if the `SchemaBasedEditorDirective` is imported. If you want to be sure you need to find it on the page. For example in `exploration_editor.html` you can see the editor here:
![image](https://user-images.githubusercontent.com/10142938/53110747-fd44d500-353b-11e9-8472-dd55cee6bae9.png)

Reply:
ok